crates/vm/levm/src/opcode_handlers/arithmetic.rs

```diff
@@ -220,45 +220,36 @@ impl VM {
         let byte_size = current_call_frame.stack.pop()?;
         let value_to_extend = current_call_frame.stack.pop()?;
 
+        if byte_size >= U256::from(31) {
+            current_call_frame.stack.push(value_to_extend)?;
+            return Ok(OpcodeSuccess::Continue);
+        }
+
         let bits_per_byte = U256::from(8);
         let sign_bit_position_on_byte = U256::from(7);
 
-        let max_byte_size = U256::from(31);
-        let byte_size = byte_size.min(max_byte_size);
-        let total_bits = bits_per_byte
+        let sign_bit_index = bits_per_byte
             .checked_mul(byte_size)
+            .and_then(|total_bits| total_bits.checked_add(sign_bit_position_on_byte))
             .ok_or(VMError::Internal(
                 InternalError::ArithmeticOperationOverflow,
             ))?;
-        let sign_bit_index =
-            total_bits
-                .checked_add(sign_bit_position_on_byte)
-                .ok_or(VMError::Internal(
-                    InternalError::ArithmeticOperationOverflow,
-                ))?;
-
-        let sign_bit_index: usize = match sign_bit_index.try_into() {
-            Ok(val) => val,
-            Err(_) => {
-                // this means the value_to_extend was too big to extend, so remains the same.
-                // Maybe this verification could be done before in this function
-                current_call_frame.stack.push(value_to_extend)?;
-                return Ok(OpcodeSuccess::Continue);
-            }
-        };
 
-        let is_negative = value_to_extend.bit(sign_bit_index);
+        let shifted_value = checked_shift_right(value_to_extend, sign_bit_index)?;
+        let is_negative = shifted_value & U256::one() == U256::one();
 
-        let sign_bit_mask = checked_shift_left(U256::one(), U256::from(sign_bit_index))?
+        if !is_negative {
+            current_call_frame.stack.push(value_to_extend)?;
+            return Ok(OpcodeSuccess::Continue);
+        };
```

Comment:
```suggestion
        if sign_bit.is_zero() {
            current_call_frame.stack.push(value_to_extend)?;
            return Ok(OpcodeSuccess::Continue);
        };
```

Reply:
Thx for the feedback! [Done](https://github.com/lambdaclass/ethrex/pull/1408/commits/c7bed6c3a2f7385e833a83a20038d7a7e4fe34ec).